test/jest/dep-graph-builders/npm-lock-v2.test.ts

```diff
@@ -56,6 +56,55 @@ describe('dep-graph-builder npm-lock-v2', () => {
         });
       });
 
+      describe.each([
+        'simple-override',
+        'simple-dotted-override',
+        'deep-override',
+      ])(
+        '[simple tests - needing strictOutOfSync=true] project: %s ',
+        (fixtureName) => {
+          it('matches expected', async () => {
+            const pkgJsonContent = readFileSync(
+              join(
+                __dirname,
+                `./fixtures/npm-lock-v2/${fixtureName}/package.json`,
+              ),
+              'utf8',
+            );
+            const pkgLockContent = readFileSync(
+              join(
+                __dirname,
+                `./fixtures/npm-lock-v2/${fixtureName}/package-lock.json`,
+              ),
+              'utf8',
+            );
+
+            const newDepGraph = await parseNpmLockV2Project(
+              pkgJsonContent,
+              pkgLockContent,
+              {
+                includeDevDeps: false,
+                includeOptionalDeps: true,
+                pruneCycles: true,
+                strictOutOfSync: true,
+              },
+            );
+
+            const expectedDepGraphJson = JSON.parse(
+              readFileSync(
+                join(
+                  __dirname,
+                  `./fixtures/npm-lock-v2/${fixtureName}/expected.json`,
+                ),
+                'utf8',
+              ),
+            );
+            const expectedDepGraph = createFromJSON(expectedDepGraphJson);
+            expect(newDepGraph.equals(expectedDepGraph)).toBeTruthy();
```

Comment:
Just because the `diff` is easier to view when this fails, I usually do

```suggestion
            expect(newDepGraph.toJSON().toEqual(expectedDepGraph.toJSON());
```

Reply:
This doesn't account for changes in shape but otherwise equivalent depgraphs though. Think shape isn't the only thing that explains equivalence.